src/vs/editor/common/viewModel/monospaceLineBreaksComputer.ts

```diff
@@ -75,22 +76,25 @@ export class MonospaceLineBreaksComputerFactory implements ILineBreaksComputerFa
 		tabSize = tabSize | 0; //@perf
 		wrappingColumn = +wrappingColumn; //@perf
 
-		let requests: string[] = [];
-		let previousBreakingData: (LineBreakData | null)[] = [];
+		const requests: string[] = [];
+		const injectedTexts: (LineInjectedText[] | null)[] = [];
+		const previousBreakingData: (LineBreakData | null)[] = [];
 		return {
-			addRequest: (lineText: string, previousLineBreakData: LineBreakData | null) => {
+			addRequest: (lineText: string, injectedText: LineInjectedText[] | null, previousLineBreakData: LineBreakData | null) => {
 				requests.push(lineText);
+				injectedTexts.push(injectedText);
 				previousBreakingData.push(previousLineBreakData);
 			},
 			finalize: () => {
 				const columnsForFullWidthChar = fontInfo.typicalFullwidthCharacterWidth / fontInfo.typicalHalfwidthCharacterWidth; //@perf
 				let result: (LineBreakData | null)[] = [];
 				for (let i = 0, len = requests.length; i < len; i++) {
+					const injectedText = injectedTexts[i];
 					const previousLineBreakData = previousBreakingData[i];
-					if (previousLineBreakData) {
+					if (previousLineBreakData && !injectedText) {
```

Comment:
We need yet another condition here `&& !previousLineBreakData.injectionTexts`. In other words, we cannot compute wrapping points incrementally if the previous or the current line contains injected text.